prime-router/src/main/kotlin/cli/tests/HistoryApiTest.kt

```diff
@@ -181,52 +180,158 @@ class HistoryApiTest : CoolTest() {
                 "${environment.url}/api/waters/org/$historyTestOrgName/submissions",
                 emptyMap(),
                 listOf("pagesize" to options.submits),
                 bearer,
                 HttpStatus.OK,
                 expectedReports = reportIds,
+                SubmissionListChecker(this),
+                doMinimalChecking = true,
             ),
-            SubmissionAPITestCase(
+            HistoryApiTestCase(
                 "no such organization",
                 "${environment.url}/api/waters/org/gobblegobble/submissions",
                 emptyMap(),
                 listOf("pagesize" to options.submits),
                 bearer,
-                HttpStatus.OK,
+                HttpStatus.UNAUTHORIZED,
                 expectedReports = emptySet(),
+                SubmissionListChecker(this),
+                doMinimalChecking = true,
             ),
         )
         if (environment != Environment.LOCAL) {
             testCases.add(
-                SubmissionAPITestCase(
+                HistoryApiTestCase(
                     "bad bearer token - TESTED ON STAGING, NOT TESTED ON LOCAL",
                     "${environment.url}/api/waters/org/$historyTestOrgName/submissions",
                     emptyMap(),
                     listOf("pagesize" to options.submits),
                     bearer + "x",
                     HttpStatus.UNAUTHORIZED,
                     expectedReports = emptySet(),
+                    SubmissionListChecker(this),
+                    doMinimalChecking = true,
                 ),
             )
         }
-        var allPassed = testCases.map {
-            val (queryPass, json) = historyApiQuery(it)
+        return runHistoryTestCases(testCases)
+    }
 
+    /**
+     * Submit a set of [testCases], parse the results, and confirm they worked.
+     * @return true if all tests passed, otherwise false.
+     */
+    fun runHistoryTestCases(testCases: List<HistoryApiTestCase>): Boolean {
+        val allPassed = testCases.map {
+            ugly("Starting test: ${it.name}")
+            val (queryPass, json) = historyApiQuery(it)
             val sanityCheck = when {
                 !queryPass -> false // api query failed.  Stop here with a fail.
                 json.isNullOrBlank() -> true // api query worked, but no json created.  Stop here with a pass.
-                else -> checkHistorySummary(it, json)
+                else -> it.jsonResponseChecker.checkJsonResponse(it, json)
             }
 
             // Only the test failed, print out the json.
-            if (!sanityCheck && !json.isNullOrBlank())
-                bad("This json failed:\n$json")
+            if (!sanityCheck && !json.isNullOrBlank()) bad("This json failed:\n$json")
+            if (sanityCheck) good("Test Passed:  ${it.name}")
             sanityCheck
         }.reduce { acc, onePassed -> acc and onePassed }
         return allPassed
     }
+}
```

Comment:
A general comment on HistoryApiTest:   this test had a number of useful utility functions to writing complex tests against the History API, namely, data objects for json responses, and the HistoryApiTestCase, and the functions that know how to run a HistoryApiTestCase.

I re-used all of this, and expanded on it too.   In particular, you can now pass in a "checker" that knows how to check various types of history results.  There are two "checkers here:  and `SubmissionListChecker` and `ReportDetailsChecker`.   Both are children of `HistoryJsonResponseChecker` .